sdks/java/harness/src/test/java/org/apache/beam/fn/harness/logging/BeamFnLoggingClientTest.java

```diff
@@ -430,21 +449,31 @@ public boolean isReady() {
       }
       // Measure how long it takes all the logs to appear.
       int sleepTime = 0;
-      while (values.size() < numEntries) {
+      while (testEntriesObserved.get() < numEntries) {
         ++sleepTime;
         Thread.sleep(1);
       }
       // Attempt to enter the blocking state by pushing back on the stream, publishing records and
       // then giving them time for it to block.
       elementsAllowed.set(false);
-      for (int i = 0; i < numEntries; ++i) {
```

Comment:
this wouldn't always trigger blocking since onReady is called every 100 rpc messages, and each message coudl be a batch of up to 1OK logs.